### PR TITLE
[Lake] Add lake tablet compaction policy

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

Automatically select base or cumulative compaction and input rowsets
according to score.

### CODE REVIEW

File: be/src/storage/lake/compaction_policy.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class Tablet;
+using TabletPtr = std::shared_ptr<Tablet>;
+class CompactionPolicy;
+using CompactionPolicyPtr = std::shared_ptr<CompactionPolicy>;
+
+// Compaction policy for lake tablet
+class CompactionPolicy {
+public:
+    virtual ~CompactionPolicy() = default;
+    virtual StatusOr<std::vector<RowsetPtr>> pick_rowsets(int64_t version) = 0;
```
Comment (sduzh) [inline be/src/storage/lake/compaction_policy.h:22]:
```suggestion

    virtual StatusOr<std::vector<RowsetPtr>> pick_rowsets(int64_t version) = 0;
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/lake/compaction_policy.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+#include "storage/lake/tablet.h"
+#include "storage/tablet_schema.h"
+
+namespace starrocks::lake {
+
+static const double kCompactionScoreThreshold = 0.999;
+
+class BaseAndCumulativeCompactionPolicy : public CompactionPolicy {
+public:
+    explicit BaseAndCumulativeCompactionPolicy(TabletPtr tablet) : _tablet(std::move(tablet)) {}
```
Comment (sduzh) [inline be/src/storage/lake/compaction_policy.cpp:16]:
```suggestion

static const double kCompactionScoreThreshold = 0.999;

class BaseAndCumulativeCompactionPolicy : public CompactionPolicy {
public:
    explicit BaseAndCumulativeCompactionPolicy(Tablet tablet) : _tablet(ablet) {}
```

File: be/src/storage/lake/compaction_policy.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    StatusOr<std::vector<RowsetPtr>> pick_rowsets(int64_t version) override;
+
+private:
+    double get_cumulative_score();
+    double get_base_score();
+    StatusOr<std::vector<RowsetPtr>> pick_cumulative_rowsets();
+    StatusOr<std::vector<RowsetPtr>> pick_base_rowsets();
+
+    TabletPtr _tablet;
```
Comment (sduzh) [inline be/src/storage/lake/compaction_policy.cpp:27]:
```suggestion
    Tablet _tablet;
```

**Review by sduzh — APPROVED** [review]

File: be/src/storage/lake/compaction_policy.cpp (hunk 3 above)
Comment (wyb) [reply be/src/storage/lake/compaction_policy.cpp:27]: Tablet raw pointer is used in `Rowset` now, if use tablet copy, the tablet pointer in `Rowset` will become nullptr when the CompactionPolicy is destructed.
Let's optimize this later.

**Pull request merged**